### PR TITLE
Build AuTest C/C++ test plugins and utilities with Automake.

### PR DESCRIPTION
**Pull request by bneradt**
Many AuTests have test plugins or test executables that are used as a
part of the test. Before this change these were built by ad hoc calls to
g++ or gcc via shell executions from within the test or one of a couple
AuTest extensions. This patch removes the building of these files from
AuTest and instead uses the Automake build system to build these. This
will build these utilities in a way that is consistent with the rest of
the system, using the same warning and error flags, the same
clang tool configurations, etc.

### CODE REVIEW

**Comment by bneradt** [comment]
[approve ci autest]

**Comment by bneradt** [comment]
[approve ci clang-analyzer]

**Comment by bneradt** [comment]
[approve ci clang-analyzer]

**Comment by masaori335** [comment]
This fixes #6438

**Comment by randall** [comment]
[approve ci autest]

**Comment by masaori335** [comment]
I'm not surprised that the clang-analyzer detects some bugs because it's the first time to run against c/c++ files in `test/`.
https://ci.trafficserver.apache.org/clang-analyzer/github/6945/2020-06-24-161846-28842-1/

**Comment by bneradt** [comment]
[approve ci clang-analyzer]

File: tests/tools/plugins/test_cppapi.cc [hunk 1]
```diff
@@ -115,9 +115,20 @@ f()
   ALWAYS_ASSERT(c2.asTSCont() != nullptr)
   ALWAYS_ASSERT(c2.mutex() == m)
 
-  ALWAYS_ASSERT(!c)
-  ALWAYS_ASSERT(c.asTSCont() == nullptr)
-  ALWAYS_ASSERT(c.mutex() == nullptr)
+  // NOLINTNEXTLINE
```
Comment (randall) [inline tests/tools/plugins/test_cppapi.cc:118]: I think clang-analyzer wants pre-processor used to hide/ignore stuff from it:

eg

#if defined(__clang_analyzer__)

#endif

https://clang-analyzer.llvm.org/faq.html#exclude_code

Comment (bneradt) [reply tests/tools/plugins/test_cppapi.cc]: Thank you Randal! Yes, I was trying to treat this as a clang-tidy issue via NOLINT comments but it was instead a clang-format one. The test passes now that I've implemented your feedback.

**Pull request merged**